client/pages/custMenu.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import Navbar from '../components/navbar';
+import RenderCustMenu from '../components/renderCustMenu';
+import AppContext from '../lib/app-context';
+import Redirect from '../components/redirect';
+
+export default class CustMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      menuFetch: [],
+      isLoading: false,
+      isModalOpen: false,
+      itemSeleted: [{ itemImg: null }]
+    };
+    this.itemClicked = this.itemClicked.bind(this);
+    this.noBtnClick = this.noBtnClick.bind(this);
+    this.yesBtnClick = this.yesBtnClick.bind(this);
+    this.qtyClicked = this.qtyClicked.bind(this);
+
+  }
+
+  componentDidMount() {
+    fetch('/api/get/menu')
+      .then(res => res.json())
+      .then(result => {
+        this.setState({ menuFetch: result, isLoading: false });
+      });
+  }
+
+  itemClicked(seletedItem) {
+    seletedItem.qty = 1;
+    this.setState({ itemSeleted: [seletedItem], isModalOpen: true });
+  }
+
+  noBtnClick(event) {
+    this.setState({ itemSeleted: [{ itemImg: null }], isModalOpen: false });
+  }
+
+  qtyClicked(event) {
+    const currentItem = this.state.itemSeleted;
+    const currentQty = parseInt(this.state.itemSeleted[0].qty);
+    if (event.target.id === 'minus') {
+      if (currentQty > 1) {
+        currentItem[0].qty = currentQty - 1;
+        this.setState({ itemSeleted: currentItem });
+      } else {
+        currentItem[0].qty = currentQty;
+        this.setState({ itemSeleted: currentItem });
+      }
+    }
+    if (event.target.id === 'plus') {
+      currentItem[0].qty = currentQty + 1;
+      this.setState({ itemSeleted: currentItem });
+    }
```

Comment:
Be careful, there is a lot of state mutation happening here. Always copy objects when making updates. 

You can make a copy like this. 

```js
const item = Object.assign({}, this.state.itemSelected);
// or 
const item = { ...this.state.itemSelected }
```